crates/proto/Cargo.toml

```diff
@@ -25,42 +25,45 @@ categories.workspace = true
 license.workspace = true
 
 [features]
-async-std = ["dep:async-std", "dep:pin-utils", "dep:socket2"]
+default = ["tokio"]
```

Comment:
I don't like making this a separate commit, because the feature is meaningless at this point, and so the additions of where it's needed are speculative. Instead, suggest adding a minimal `std` feature with the `core`/`std` changes for the `net` imports, where there is a clear impact of the feature, and ideally bundle the CI changes in the same commit to make it more obvious that the goal is MSRV compatibility.

(The `map` commit can come after that.)

Reply:
Note all the other feedback too.

Reply:
Also added the CI changes in. Let me know if there's anything else I didn't do

Reply:
I asked for a **minimal** `std` feature. There's no point (with the current scope of the feature, limited to the map changes and the `net` imports) to have all the other features enable the `std` feature.

Reply:
Sorry, I don't fully follow, could you please (re)explain what your idea for this PR is like I'm five?

Reply:
- One commit that does the minimal amount of change to optionally enable `core::net`, guarded by a `std` feature
- One commit that replaces `HashMap` with `BTreeMap` based on that same `std` feature
- Make sure that CI tests everything that it tests today for MSRV
- Additionally test that the non-`std` path in -proto is tested on stable

Reply:
Apart from "Additionally test that the non-std path in -proto is tested on stable", isn't everything else already as it is right now? What's the diff?
- We have two commits, one enables core::net
- one commit for the maps
- everything is tested in CI as it is today
- (?)

Reply:
The `Cargo.toml` changes add a bunch of dependencies of other features on `std` which are unnecessary as of this PR.

Reply:
I think specifically for the -proto crate (which few users likely depend on directly) that is fine, especially since we're doing a semver-incompatible bump.

Reply:
Your call